src/main/java/com/cleanroommc/modularui/widgets/textfield/TextFieldWidget.java

```diff
@@ -95,7 +95,7 @@ protected void setupDrawText(ModularGuiContext context, WidgetTextFieldTheme wid
         this.renderer.setSimulate(false);
         this.renderer.setPos(getArea().getPadding().left, 0);
         this.renderer.setScale(this.scale);
-        this.renderer.setAlignment(this.textAlignment, -1, getArea().height);
+        this.renderer.setAlignment(this.textAlignment, getArea().width - (getArea().getPadding().left * 2), getArea().height);
```

Comment:
You can not assume that left and right padding is equal. Use `padding.horizontal()`